### DISCUSSION TITLE
Options to disable the fastSeek

### DISCUSSION THREAD

**Discussion by amtins** [videojs/video.js#7561] [Ideas]
Videojs [v7.9.0](https://github.com/videojs/video.js/releases/tag/v7.9.0) introduced the `fastSeek`, it's a great feature. However, when a media has `chapters`, this results in starting the playback a few seconds before the beginning of the chapter. On our use case we had to do something like `player.tech('Html5').el().fastSeek = undefined;` to avoid having a weird behavior, which is a bit sad 😢

One could imagine having a property to disable the `fastSeek` like :
``` javaScript
const player = new videojs('#p', {
  html5: {
    fastSeek : 'yes', 'no', 'never again', 'please NO', 'i don\'t know', 'why not' // true or false will do the trick too
  }
});
```

In the `Html5` tech :
```javaScript
class Html5 extends Tech {
  constructor(player, options){
    //...
   const {fastSeek} = options;
   this.isFastSeekEnabled_ = fastSeek && this.el_.fastSeek && browser.IS_ANY_SAFARI;
  }
 // ...
  setCurrentTime(seconds) {
    try {
      if (this.isScrubbing_ && this.isFastSeekEnabled_) {
        this.el_.fastSeek(seconds);
      } else {
        this.el_.currentTime = seconds;
      }
    } catch (e) {
      log(e, 'Video is not ready. (Video.js)');
      // this.warning(VideoJS.warnings.videoNotReady);
    }
  }
}
```

**Comment by gkatsev** [comment]
I did recently fix it so that `fastSeek` is only used when dragging around in the progress bar and not just a regular click to seek: https://github.com/videojs/video.js/pull/7527

Is that sufficient for your needs?

**Comment by amtins** [comment]
@gkatsev thanks for your reply 👍. I actually saw the fix #7527 back when I wrote my proposal. However, we have a slightly more complex use case for legal reasons. We use [middleware](https://docs.videojs.com/tutorial-middleware.html) that takes care of blocking certain portions of a media when the user seeks which leads to unwanted effects.

More generally, having the ability to disable `fastSeek` would allow for a unified experience across all browsers where precision is prioritized over a fast seek  a bit like the [overrideNative](https://github.com/videojs/http-streaming#overridenative).
